### ISSUE TITLE
“Change subscribed languages” doesn’t work properly 

### ISSUE THREAD

**Issue by DianaiD** [mastodon/mastodon#21722]
### Steps to reproduce the problem

1. I follow somebody and then choose to subscribe only one language
2. I this account shares some ohter post in language what I didn’t subscribe I still see it :(
3.
...

### Expected behaviour

I will see only content in language what I choose in my homefeed

### Actual behaviour

I still see posts in wrong language in my newsfeed if they are shared by someone

### Specifications

Mastodon 4.0.2 - it is not related with my op system (but I use iPad with Safari if this help you ;) )
![1A76B214-C670-4009-A84C-A700DDB76F44](https://user-images.githubusercontent.com/797612/204127060-f807dc01-da45-4fe2-b487-eb87f11fce03.jpeg)
![FD109D44-9161-4511-A4F8-F4AB4032195D](https://user-images.githubusercontent.com/797612/204127062-507ec995-7e76-4f89-899a-fdb9e6db2ecd.jpeg)
![E99F42F6-D679-4268-ACF0-2A3CF0D2BA18](https://user-images.githubusercontent.com/797612/204127065-18a2f3cd-63a3-43a9-9092-08c06a877b4e.jpeg)

**Comment by jotaemei** [comment]
Yes, I can confirm this bug. the subscriptions language feeds are showing false positives. Some may suspect that this is because posts were actually tagged with a different language than they're written in, but I've tested it through a couple accounts I have and where the posts are tagged correctly in different languages. They're not getting filtered out.
